pkg/v1/builder/command/cli_compile.go

```diff
@@ -199,25 +128,40 @@ type errInfo struct {
 	ID   string
 }
 
-func compile(cmd *cobra.Command, args []string) error {
-	if version == "" {
-		log.Fatal("version flag must be set")
+// setGlobals initializes a set of global variables used throughout the compile
+// process, based on the arguments passed in.
+func setGlobals(compileArgs *PluginCompileArgs) {
+	if compileArgs.Version == "" {
+		log.Fatal("version value must be set")
 	}
-	log.Infof("building local repository at ./%s", artifactsDir)
+
+	version = compileArgs.Version
+	artifactsDir = compileArgs.ArtifactsDir
+	ldflags = compileArgs.LDFlags
+	tags = compileArgs.Tags
+	goprivate = compileArgs.GoPrivate
+	targetArch = compileArgs.TargetArch
+}
+
+func Compile(compileArgs *PluginCompileArgs) error {
+	// Set our global values based on the passed args
+	setGlobals(compileArgs)
+
+	log.Infof("building local repository at ./%s", compileArgs.ArtifactsDir)
```

Comment:
nit: I think you can get rid of the `./` prefix as artifacts directory passed is already of the form `./artifacts`.

Reply:
Sure, will update.